include/nil/blueprint/extract_constructor_parameters.hpp

```diff
@@ -0,0 +1,97 @@
+//---------------------------------------------------------------------------//
+// Copyright (c) 2022 Mikhail Komarov <nemo@nil.foundation>
+// Copyright (c) 2022 Nikita Kaskov <nbering@nil.foundation>
+//
+// MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//---------------------------------------------------------------------------//
+
+#ifndef CRYPTO3_ASSIGNER_EXTRACT_CONSTRUCTOR_PARAMETERS_HPP
+#define CRYPTO3_ASSIGNER_EXTRACT_CONSTRUCTOR_PARAMETERS_HPP
+
+namespace nil {
+    namespace blueprint {
+        namespace detail {
+
+            template <typename BlueprintFieldType>
+            typename BlueprintFieldType::value_type extract_component_constructor_parameter_field(llvm::Value *input_value){
```

Comment:
As I see, the functions in this file are not specific to `component`, `constructor`, `parameter`, or `intrinsic`. They just extract values from a constant. Maybe rename them to something like `extract_constant_field_value` and so on?

Reply:
This is constants that are component constructor parameters and must not be written into assignment table. But also we have constant column, maybe in the future we will work with the second type of constants field values and confusion may occur. This way of naming distinguishes two cases